app/src/main/java/com/duckduckgo/app/browser/favicon/FaviconPersister.kt

```diff
@@ -118,14 +120,22 @@ class FileBasedFaviconPersister(
         val faviconFile = prepareDestinationFile(directory, subFolder, domain)
         writeBytesToFile(faviconFile, bitmap)
 
-        return faviconFile
+        return if (faviconFile.exists()) {
+            faviconFile
+        } else {
+            null
+        }
     }
 
     @Synchronized
     private fun writeBytesToFile(file: File, bitmap: Bitmap) {
-        FileOutputStream(file).use { outputStream ->
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream)
-            outputStream.flush()
+        try {
+            FileOutputStream(file).use { outputStream ->
+                bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream)
+                outputStream.flush()
+            }
+        } catch (e: FileNotFoundException) {
```

Comment:
As discussed, let's catch all exceptions here since we don't any problems to cause a crash

Reply:
Done!